src/actions/VaccineActions.js

```diff
@@ -25,6 +32,73 @@ const blinkStop = () => ({ type: VACCINE_ACTIONS.BLINK_STOP });
 const scanStart = () => ({ type: VACCINE_ACTIONS.SCAN_START });
 const scanStop = () => ({ type: VACCINE_ACTIONS.SCAN_STOP });
 const sensorFound = macAddress => ({ type: VACCINE_ACTIONS.SENSOR_FOUND, payload: { macAddress } });
+const downloadLogsStart = () => ({
+  type: VACCINE_ACTIONS.DOWNLOAD_LOGS_START,
+});
+const downloadLogsError = () => ({
+  type: VACCINE_ACTIONS.DOWNLOAD_LOGS_ERROR,
+});
+const downloadLogsComplete = () => ({
+  type: VACCINE_ACTIONS.DOWNLOAD_LOGS_COMPLETE,
+});
+
+const downloadAllLogs = () => async dispatch => {
+  dispatch(downloadLogsStart());
+
+  // Ensure there are some sensors which have been assigned a location before syncing.
+  const sensors = UIDatabase.objects('Sensor').filtered('location != null && isActive == true');
+
+  if (!sensors.length) {
+    // TODO: Should we do something if there are errors?
```

Comment:
Could add a Toast notification for now - though I think not needed for log downloading. If there are no sensors then there's nothing to do.

Reply:
I thought about this more/again and I think it'd be better to not do my suggestion... 🤣

Reply:
I was thinking about this a bit too and think we may need to go back to the temperature log table anyway to fetch the last sync timestamp per sensor? There will likely be a period of time when a user first launches the app and won't know the last sync time until a successful one is done (I think? If my understanding of the way people use the app is correct - unless people just never close the app and tend to keep it running? Then this doesn't apply).

Maybe something like on start, fetch last temp logs and calculate/guess timestamps based on the record, then after each successful log download, update the timestamp/sync state? 

I also wanted to add some 🍞 🔥 to the UI  but I couldn't think of a way that would be helpful rather than annoying haha, I agree a failure indicator somewhere might be nice. Could that be an idea for when to turn a sensor orange/red on the sensor page? https://marvelapp.com/prototype/55aea8f/screen/75648034. Or turn it into a sync fail icon?? 🤷

Reply:
You're right about determining the last sync date. Also good idea regarding the circle: The last sync failing would be easier as state (which we can persist between app starts, if needed, kind of like a database!) I'm not sure exactly where the state should live, though. Maybe a new issue? Can determine exactly what state should be stored regarding the downloading of temperatures?

Reply:
Cool yeah, I think best for a new issue, have created #3380 for it. That's good to know about persisting between app starts!